File: src/model/Listing.java
package model;

/**
 * Model representing a real estate listing in the Java service.
 */
public class Listing {
    public String id;
    public String title;
    public double price;
    public String gender;
    public String roomType;
    public String propertyType;
    
    // Location parameters
    public double distanceFromCampus;
    public String area;

    public Listing() {}
}

File: egypt_real_estate/dashboard/models/Listing.js
const mongoose = require('mongoose');

const priceSchema = new mongoose.Schema({
  amount_egp: Number,
  period:     String,
  raw:        String,
}, { _id: false });

const listingSchema = new mongoose.Schema({
  source:        String,
  ad_id:         String,
  prop_id:       String,
  url:           String,
  title:         String,
  area:          String,
  prop_category: String,
  page:          Number,
  price:         priceSchema,
  property_type: String,
  bedrooms:      Number,
  bathrooms:     Number,
  area_sqm:      Number,
  furnished:     mongoose.Schema.Types.Mixed,
  compound:      String,
  location:      String,
  posted_ago:    String,
  listed_date:   String,
  is_verified:   Boolean,
  scraped_at:    String,
}, { timestamps: true });

module.exports = mongoose.model('Listing', listingSchema);
